docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -0,0 +1,535 @@
+
+The ``User`` type extends ``AuditLog`` and inherits the ``created_at`` property as a result. This property is auto-filled by the abstract type via the ``datetime_current`` function. The datetime is saved as a UTC timestamp. Type User also has an annotation field. Annotations allow us to attach arbitrary descriptions to the types. Along with the inherited type, the user type also defines a concrete required property called ``name``. We impose two constraints on this property—names should be unique and they can't be longer than 50 characters.
+
+Similar to the ``User`` type, we define an ``Event`` type that extends the ``AuditLog`` abstract type. An event will also have a ``name`` property and a few additional concrete properties like ``address`` and ``schedule``. While ``address`` holds string data, ``schedule`` expects the incoming data to be formatted as datetime. An ``Event`` can also have an optional link to a ``User``. This user here represents the host of an event. Currently, we're only allowing a single host to be attached to an event.
+
+
+Build the API endpoints
+=======================
+
+The API endpoints are defined in the `app` directory. The directory structure looks as follows:
+
+::
+    app
+    ├── __init__.py
+    ├── events.py
+    ├── main.py
+    └── users.py
+
+The `user.py` and `event.py` modules contain the code to build the ``User`` and ``Event`` APIs respectively. The ``main.py`` module then registers all the endpoints and exposes them to the `uvicorn <https://www.uvicorn.org/>`_ webserver.
+
+
+User APIs
+^^^^^^^^^^
+
+Since the ``User`` type is simpler among the two, we'll start with that. Let's create a `GET users/` endpoint so that we can start looking at the user objects saved in the database. You can create the API with a couple of lines of code in FastAPI:
+
+.. code-block:: python
+    # fastapi-crud/app/users.py
+
+    from __future__ import annotations
+
+    import datetime
+    from http import HTTPStatus
+    from typing import Iterable
+
+    import edgedb
+    from fastapi import APIRouter, HTTPException, Query
+    from pydantic import BaseModel
+
+    router = APIRouter()
+    client = edgedb.create_async_client()
+
+
+    class RequestData(BaseModel):
+        name: str
+
+
+    class ResponseData(BaseModel):
+        name: str
+        created_at: datetime.datetime
+
+
+    @router.get("/users")
+    async def get_users(
+        name: str = Query(None, max_length=50)
+        ) -> Iterable[ResponseData]:
+
+        if not name:
+            users = await client.query("SELECT User {name, created_at};")
+        else:
+            users = await client.query(
+                """SELECT User {name, created_at} FILTER User.name=<str>$name""",
+                name=name,
+            )
+        response = (
+            ResponseData(name=user.name,
+            created_at=user.created_at) for user in users
+        )
```

Comment:
This formatting style is a bit easier to read.

```suggestion
        response = (
            ResponseData(
                name=user.name,
                created_at=user.created_at
            ) for user in users
        )
```

It's rather unfortunate that we have to do this repack manually. We need to improve the situation here asap, by providing a way to generate these pydantic models automatically for queries.